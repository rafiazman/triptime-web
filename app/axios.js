/** @format */

import axios from 'axios';

/**
 * Creates an instance of axios pre-configured with settings
 * to interact with backend SPA authentication system
 * @returns {axios}
 */
export default axios.create({
  baseURL: `${process.env.API_HOSTNAME}/api`,
  withCredentials: true,
  responseType: 'json',
});
