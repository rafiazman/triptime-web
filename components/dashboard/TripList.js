/** @format */

import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import TripCard from './TripCard';
import styles from '../../css/homepage.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ReactLoading from 'react-loading';

export default class TripList extends React.Component {
  render() {
    const tripInfoList = this.props.tripInfoList;
    return (
      <div className={styles.component}>
        <h2>
          {' '}
          <FontAwesomeIcon icon={this.props.icon} /> {this.props.title}{' '}
        </h2>
        {this.props.loading ? (
          <div className={styles.loadingContainer}>
            <ReactLoading type='bubbles' color='#ff4200' />
          </div>
        ) : tripInfoList.length > 0 ? (
          <div>
            {tripInfoList.map((tripInfo, index) => (
              <Link href={`/trip/${tripInfo.id}`} key={index}>
                <a>
                  <TripCard
                    tripName={tripInfo.name}
                    updated={tripInfo.updated}
                  />
                </a>
              </Link>
            ))}
          </div>
        ) : (
          this.props.displayIfNoTrip
        )}
      </div>
    );
  }
}

TripList.propTypes = {
  displayIfNoTrip: PropTypes.element.isRequired,
  icon: PropTypes.object.isRequired,
  tripInfoList: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
};
