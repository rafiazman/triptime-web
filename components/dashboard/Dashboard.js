/** @format */

import React from 'react';
import PropTypes from 'prop-types';
import SiteInfo from '../SiteInfo';
import Link from 'next/link';
import Greeting from '../Greeting';
import styles from '../../css/homepage.module.css';
import TripList from './TripList';
import axios from 'axios';

import {
  faShoePrints,
  faPen,
  faClock,
} from '@fortawesome/free-solid-svg-icons';

export default class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      pastTrips: [],
      currentTrips: [],
      planningTrips: [],
      planningLoading: true,
      currentLoading: true,
      pastLoading: true,
    };
  }

  componentDidMount() {
    const hostName = process.env.API_HOSTNAME;
    axios
      .get(`${hostName}/api/trips/current`)
      .then(
        response => response.data,
        () => [],
      )
      .then(trips =>
        this.setState(() => ({ currentTrips: trips, currentLoading: false })),
      );
    axios
      .get(`${hostName}/api/trips/past`)
      .then(
        response => response.data,
        () => [],
      )
      .then(trips =>
        this.setState(() => ({ pastTrips: trips, pastLoading: false })),
      );
    axios
      .get(`${hostName}/api/trips/future`)
      .then(
        response => response.data,
        () => [],
      )
      .then(trips =>
        this.setState(() => ({ planningTrips: trips, planningLoading: false })),
      );
  }

  render() {
    const currentTrips = this.state.currentTrips;
    const pastTrips = this.state.pastTrips;
    const planningTrips = this.state.planningTrips;
    return (
      <div className={styles.page}>
        <div className={styles.content}>
          <Greeting name={this.props.name} />
          <TripList
            icon={faShoePrints}
            title='Your Current Trips:'
            tripInfoList={currentTrips}
            displayIfNoTrip={<h3>Your next adventure is yet to come...</h3>}
            loading={this.state.currentLoading}
          />

          <TripList
            icon={faPen}
            title='Your Planned Trips:'
            tripInfoList={planningTrips}
            displayIfNoTrip={
              <h3>
                <span style={{ marginRight: '5px' }}>No plans yet.</span>
                <Link href='/newtrip'>
                  <a>Plan for a trip</a>
                </Link>
                <span style={{ marginLeft: '5px' }}>today!</span>
              </h3>
            }
            loading={this.state.planningLoading}
          />
          <TripList
            icon={faClock}
            title='Your Memories: '
            tripInfoList={pastTrips}
            displayIfNoTrip={
              <h3>
                No past trips yet. What memory will you create at TripTime?
              </h3>
            }
            loading={this.state.pastLoading}
          />
        </div>
        <SiteInfo />
      </div>
    );
  }
}

Dashboard.propTypes = {
  name: PropTypes.string.isRequired,
};
