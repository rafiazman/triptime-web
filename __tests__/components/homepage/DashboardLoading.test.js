/** @format */

import TestRenderer from 'react-test-renderer';
import React from 'react';
import Dashboard from '../../../components/dashboard/Dashboard';
import TripList from '../../../components/dashboard/TripList';
import axios from 'axios';

describe('Test Dashboard Loading', () => {
  const DATE_TO_USE = new Date(2020, 11, 17, 23, 24, 0);
  const _Date = Date;
  global.Date = jest.fn(() => DATE_TO_USE);
  global.Date.UTC = _Date.UTC;
  global.Date.parse = _Date.parse;
  global.Date.now = _Date.now;
  axios.get.mockResolvedValue({ data: [] });
  const userHomePageRenderer = TestRenderer.create(
    <Dashboard name={'Tester'} />,
  );
  test('Check if Dashboard renders three TripList', () => {
    expect(userHomePageRenderer.root.findAllByType(TripList).length).toBe(3);
  });
  test('Check if Dashboard displays three loading bars when the trips are loading', () => {
    userHomePageRenderer.root.instance.setState({
      currentLoading: true,
      pastLoading: true,
      planningLoading: true,
    });
    expect(userHomePageRenderer.toJSON()).toMatchSnapshot();
  });
});
