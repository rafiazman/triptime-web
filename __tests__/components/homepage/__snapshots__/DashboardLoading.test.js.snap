// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard Loading Check if Dashboard displays three loading bars when the trips are loading 1`] = `
<div
  className="page"
>
  <div
    className="content"
  >
    <h1>
      Tester
      , good night
    </h1>
    <div
      className="component"
    >
      <h2>
         
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-shoe-prints fa-w-20 "
          data-icon="shoe-prints"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 160h32V32h-32c-35.35 0-64 28.65-64 64s28.65 64 64 64zM0 416c0 35.35 28.65 64 64 64h32V352H64c-35.35 0-64 28.65-64 64zm337.46-128c-34.91 0-76.16 13.12-104.73 32-24.79 16.38-44.52 32-104.73 32v128l57.53 15.97c26.21 7.28 53.01 13.12 80.31 15.05 32.69 2.31 65.6.67 97.58-6.2C472.9 481.3 512 429.22 512 384c0-64-84.18-96-174.54-96zM491.42 7.19C459.44.32 426.53-1.33 393.84.99c-27.3 1.93-54.1 7.77-80.31 15.04L256 32v128c60.2 0 79.94 15.62 104.73 32 28.57 18.88 69.82 32 104.73 32C555.82 224 640 192 640 128c0-45.22-39.1-97.3-148.58-120.81z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Your Current Trips:
         
      </h2>
      <div
        className="loadingContainer"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <circle transform=\\"translate(8 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(16 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.3\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(24 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#ff4200",
              "height": 64,
              "width": 64,
            }
          }
        />
      </div>
    </div>
    <div
      className="component"
    >
      <h2>
         
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-pen fa-w-16 "
          data-icon="pen"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Your Planned Trips:
         
      </h2>
      <div
        className="loadingContainer"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <circle transform=\\"translate(8 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(16 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.3\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(24 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#ff4200",
              "height": 64,
              "width": 64,
            }
          }
        />
      </div>
    </div>
    <div
      className="component"
    >
      <h2>
         
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Your Memories: 
         
      </h2>
      <div
        className="loadingContainer"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <circle transform=\\"translate(8 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(16 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.3\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(24 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#ff4200",
              "height": 64,
              "width": 64,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <h1>
        <span
          className="logo-span"
        >
          <img
            alt="TripTime logo"
            className="logo"
            src="/img/logo.svg"
          />
          <span>
            TripTime
          </span>
        </span>
      </h1>
      <div
        className="links"
      >
        <h2
          className="not-on-small-screen"
        >
          TripTime
        </h2>
        <p>
          <a>
            Home
          </a>
        </p>
        <p>
          <a>
            Features
          </a>
        </p>
        <p>
          <a>
            About Us
          </a>
        </p>
        <p>
          <a
            href="https://github.com/tantigers/TripTime"
          >
            Code on Github
          </a>
        </p>
      </div>
      <div
        className="follow"
      >
        <h2>
          Follow Us
        </h2>
        <div
          className="icons"
        >
          <a
            href="https://github.com/tantigers"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-github fa-w-16 "
              data-icon="github"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            href="https://www.youtube.com/"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-youtube fa-w-18 "
              data-icon="youtube"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            href="https://www.facebook.com/"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-facebook fa-w-16 "
              data-icon="facebook"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
      </div>
      <p
        className="copyright"
      >
        ©2020 TripTime All Rights Reserved.
      </p>
    </div>
  </div>
</div>
`;
